### PR TITLE
go upgrade

### PR DESCRIPTION
**Pull request by weilirs**
**Description**

This PR fixes https://github.com/meshery/meshery/issues/12829

**Notes for Reviewers**
`secrets.GO_VERSION` needs to be upgraded as well

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by weilirs** [comment]
@leecalcote 
To test it with Meshery server, is it to run `go build ./...` and add `replace github.com/layer5io/meshkit => /path/to/local/meshkit` to the Meshery server and then run `go test`?

**Comment by weilirs** [comment]
I ran `go build ./...` in Meshkit.
Then I added this line `github.com/layer5io/meshkit => /Users/macbookpro/Documents/meshkit` to the Meshery server's `go.mod`'s "replace" clause.
Then I ran `make server` and check `http://localhost:9081`. It is working fine now.

**Comment by leecalcote** [comment]
@weilirs good work.

Do you know why we're seeing these build check failures? https://github.com/meshery/meshkit/actions/runs/12401032586/job/34619527275?pr=624#step:4:28

**Comment by weilirs** [comment]
> @weilirs good work.
> 
> Do you know why we're seeing these build check failures? [meshery/meshkit/actions/runs/12401032586/job/34619527275?pr=624#step:4:28](https://github.com/meshery/meshkit/actions/runs/12401032586/job/34619527275?pr=624#step:4:28)

Let me check this out

**Comment by weilirs** [comment]
> @weilirs good work.
> 
> Do you know why we're seeing these build check failures? https://github.com/meshery/meshkit/actions/runs/12401032586/job/34619527275?pr=624#step:4:28

`golangci-lint-action` and `make check` both run `golangci-lint` so I added `skip-cache = true` to skip the second lint run
And there are some unused variables and function so I added `//nolint:unused` to temporarily skip them

**Comment by weilirs** [comment]
It seems most of the recent PRs fail this check
```
--- FAIL: TestValidator (0.01s)

    validate_test.go:70: unknown domain for reference "../core.json#/definitions/uuid" (and 4 more errors)

    validate_test.go:70: unknown domain for reference "../core.json#/definitions/uuid" (and 4 more errors)

    validate_test.go:80: test failed for catalog_data, got true, want true error: <nil>

    validate_test.go:70: unknown domain for reference "../core.json#/definitions/uuid" (and 4 more errors)

    validate_test.go:80: test failed for models, got true, want true error: <nil>

FAIL

coverage: 67.4% of statements

FAIL	github.com/layer5io/meshkit/validator	0.050s

FAIL

make: *** [Makefile:10: test] Error 1
```

**Comment by leecalcote** [comment]
@weilirs, please be aware that a merge conflict has arisen. Will you have a look into resolving this conflict?

**Comment by leecalcote** [comment]
@devhindo, will you give this PR a review?

**Comment by weilirs** [comment]
> @weilirs, please be aware that a merge conflict has arisen. Will you have a look into resolving this conflict?

I will

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,13 +15,14 @@ jobs:
       - uses: actions/checkout@master
       - uses: actions/setup-go@master
         with:
-          go-version: '1.21'
+          go-version: '1.23'
           cache: true
           cache-dependency-path: go.sum
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v4.0.0
```
Comment (devhindo) [inline .github/workflows/ci.yml:22]: update `golangci/golangci-lint-action@v4.0.0` to `golangci/golangci-lint-action@v6`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -44,7 +45,7 @@ jobs:
       - uses: actions/checkout@master
       - uses: actions/setup-go@v4
```
Comment (devhindo) [inline .github/workflows/ci.yml:46]: update `actions/setup-go@v4` to `actions/setup-go@v5`

**Review by devhindo — CHANGES_REQUESTED** [review]
also, did you make sure that you ran `go mod tidy`?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (weilirs) [reply .github/workflows/ci.yml]: updated

File: .github/workflows/ci.yml (hunk 2 above)
Comment (weilirs) [reply .github/workflows/ci.yml]: updated

**Comment by weilirs** [comment]
> also, did you make sure that you ran `go mod tidy`?

I made the changes you suggested and then ran `go mod tidy` and nothing return. 
And I ran `go mod tidy` before as well.

**Comment by weilirs** [comment]
sorry I messed up the sign-off, let me close this and open a new PR.

**Pull request closed without merging**